webxrlayers-1.bs

```diff
@@ -1386,6 +1387,64 @@ when the [=XR Compositor=] can no longer reproject the layer.
 
+Hit testing {#hittesting}
+===========
+
+XRLayerHitResult {#xrlayerhitresulttype}
+----------------
+
+<pre class="idl">
+interface XRLayerHitResult {
+  readonly attribute XRCompositionLayer layer;
+  readonly attribute unsigned long x;
+  readonly attribute unsigned long y;
+  readonly attribute XRSpace hitSpace;
+};
+</pre>
+
+The <dfn attribute for="XRLayerHitResult">layer</dfn> member returns what layer was hit.
+
+The <dfn attribute for="XRLayerHitResult">x</dfn> and <dfn attribute for="XRLayerHitResult">y</dfn> member return the x and y position
+where the layer was hit. This position is <code>0</code for the top/left position and <code>1</code> for the bottom/right position.
+
+The <dfn attribute for="XRLayerHitResult">position</dfn> member returns the position in 3D space where the hit occured.
+
+Finding hit tests {#finding-hit-tests}
+-----------------
+
+<pre class="idl">
+[SecureContext, Exposed=Window] partial interface XRFrame {
+  sequence&lt;XRLayerHitResult&gt; findHitTest(XRSpace space, XRRigidTransform ray);
+};
+</pre>
+
+<div class="algorithm" data-algorithm="findHitTestAlgo">
+
+The <dfn method for="XRFrame">findHitTest(XRSpace |space|, XRRigidTransform |ray|)</dfn> method casts a ray into the
+scene containing instances of {{XRCompositionLayer|XRCompositionLayers}}.
+
+When this method is invoked, the user agent MUST run the following steps:
+  1. If [=this=] {{XRFrame/session}} was not created with "[=feature descriptor/layers=]" enabled, throw an {{NotSupportedError}} and abort these steps.
+  1. If [=this=]'s [=XRFrame/active=] boolean is <code>false</code>, throw an {{InvalidStateError}} and abort these steps.
+  1. Let |hitresults| be a new sequence created in the [=relevant realm=] of [=this=].
+  1. For each |layer| in [=this=] {{XRFrame/session}}'s {{XRRenderState/layers}}:
+    1. If |ray| using the orign |space| doesn't intersect |layer|, continue to the next entry.
+    1. Let |result| be a [=new=] instance of {{XRLayerHitResult}} in the [=relevant realm=] of [=this=].
+    1. Set |result|'s {{XRLayerHitResult/layer}} to the hit {{XRCompositionLayer}}.
+    1. Set |result|'s {{XRLayerHitResult/x}} and {{XRLayerHitResult/y}} to the position where the |ray| hit {XRLayerHitResult/layer}}.
+    1. Initialize |result|'s {{XRLayerHitResult/hitSpace}} as follows:
+        <dl class="switch">
+          <dt> If a previous hit was found with the same |layer| and [=this=], the user agent MAY:
+            <dd> Let |result|'s {{XRLayerHitResult/hitSpace}} be the same {{XRSpace}} object as returned by an earlier call with the same |layer| and [=this=].
+          <dt> Otherwise
+            <dd> Let |result|'s {{XRLayerHitResult/hitSpace}} be a [=new=] {{XRSpace}} in the [=relevant realm=] of [=this=].
+        </dl>
+    1. Set |result|'s {{XRLayerHitResult/hitSpace}} to an {{XRSpace}} that has a [=XRSpace/native origin=] to the pose that should be used to render the 3D position that was hit.
```

Comment:
This should be defined above, and it should be mentioned that  the space is only valid during frames where the hit has happened.

Reply:
> This should be defined above

What do you mean by "defined above"?

> it should be mentioned that the space is only valid during frames where the hit has happened.

I looked how `gripSpace` is defined in the WebXR spec but I don't see where it mentions that it's only valid during the XRFrame.